### DISCUSSION TITLE
What is a repository

### DISCUSSION THREAD

**Discussion by git-hls-demo** [community/community#21978] [New to GitHub]
<p>Hi,</p>
<p>I am a bit confused with what exactly a repository is.<br>
Is it the .git-folder or is it the folder containing the files to be version controlled and .git?<br>
In other words, do the files to be version controlled live inside the repository or does it contain only the history/changes?</p>
<p>Thanks!</p>

**Comment by byrneh** [comment]
<p>Maybe this explains for you?<br>
<a href="https://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository" rel="noopener nofollow ugc">Git-Basics-Getting-a-Git-Repository</a></p>
<p>GitHub has some excellent free learning labs I would highly recommend if you are looking to get started using GitHub<br>
<a href="https://lab.github.com/githubtraining/introduction-to-github" rel="noopener nofollow ugc">introduction-to-github</a></p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="git-hls-demo" data-post="1" data-topic="246920">
<div class="title">
<div class="quote-controls"></div>
 git-hls-demo:</div>
<blockquote>
<p>Is it the .git-folder or is it the folder containing the files to be version controlled and .git?</p>
</blockquote>
</aside>
<p>Both, kind of. The <code>.git/</code> directory contains the repository data, the history and all that. “History” includes all versions of all files that were ever committed. The files you work on (in the directory that also contains the <code>.git/</code> directory) are called the <em>working tree</em>, where you prepare things to be committed. It is possible to have a repository without a working tree (that’s called a <em>bare repository</em>), but that’s only useful for things like hosting (like what GitHub does) or backups.</p>
<p>If you want to understand what’s going on inside <code>.git/</code>, take a look at <a href="https://git-scm.com/book/en/v2/Git-Internals-Plumbing-and-Porcelain" class="inline-onebox">Git - Plumbing and Porcelain</a> and the following sections.</p>

**Comment by git-hls-demo** [comment]
<p>This was very helpful, thank you.</p>

**Comment by queenofcorgis** [comment]
<p>Welcome to the community, we’re so glad you’re here! As a reminder, if you get an answer to your question, please mark that answer as the solution for future members who may be wondering the same thing!</p>
